### DISCUSSION TITLE
Connect Kafka Topic to Streampipes Message Structure

### DISCUSSION THREAD

**Discussion by pm60lixy** [apache/streampipes#1951] [Q&A]
Hey, 
 i try to connect a Kafka Topic to Streampips but i get problems with the strcuture of the kafka messages. I have already tried different formats of the kafka messages, but streampipes either does not recognize the schema at all or only the variable fields without the payload. Can anybody help me with this problem? whats the correct structure for this type of kafka messages? 

<img width="1219" alt="Nachrichten-Schema-1" src="https://github.com/apache/streampipes/assets/102787963/8253ecf5-3e49-426f-8760-185e2b0adfc0">
<img width="1161" alt="Nachrichten-Schema-2" src="https://github.com/apache/streampipes/assets/102787963/a74b3e0a-5c98-43f4-bb5f-d8c3d3a63904">
<img width="802" alt="Nachrichten-Schema-3" src="https://github.com/apache/streampipes/assets/102787963/2f212b93-7692-4df8-9ff2-0a666e8da74f">
<img width="1160" alt="Bildschirmfoto 2023-06-05 um 15 04 45" src="https://github.com/apache/streampipes/assets/102787963/313d566e-4da9-4853-85d5-439f33d8f92d">
<img width="461" alt="Bildschirmfoto 2023-06-05 um 15 06 45" src="https://github.com/apache/streampipes/assets/102787963/25fd80fd-ff29-44ae-94b1-ad091057feff">

best regards

**Comment by tenthe** [comment]
Hi @pm60lixy,
thanks for opening the issue.
Do I understand you correctly, you plan to read the values in the "payload" field?
Therefore, you should use the "Json Object" format and then edit the event schema in the editor.

When does the above error occur? When you try to guess the event schema or when you start an adapter?
Do all events in this theme have the same schema or can they be different?

Cheers,
Philipp

**Reply by obermeier** [reply]
I believe the error is occurring due to an attempt to guess the data type of a null value. 
You can find a discussion on this topic here #860.
